language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/BallerinaCompletionContextImpl.java

```diff
@@ -82,4 +87,30 @@ public void addResolver(Node node) {
     public List<Node> getResolverChain() {
         return this.resolverChain;
     }
+
+    @Override
+    public Optional<TypeSymbol> getContextType() {
+
+        if (this.isContextTypeFound) {
+            return this.contextType;
+        }
+
+        this.isContextTypeFound = true;
+        if (this.getNodeAtCursor() == null) {
+            this.contextType = Optional.empty();
+            return this.contextType;
+        }
+
+        NonTerminalNode node = getNodeAtCursor();
+        do {
```

Comment:
Is there any specific reason to use a loop here? The parent visit should be carry out by the visitor itself. Hence we can avoid using the loop here and depend on the `ContextTypeResolver`'s behaviour

Reply:
There can be instances where the parent is not visited. The ExternalTreeList Node implementation is one such instance. (The apply method does not visit the transformer). Hense, the parent will not be visited. IMO, to handle such scenarios a loop is required since we can't guarantee that the ContextTypeResolver always resolves a context for a given node

Reply:
+1. Let's proceed with this approach then. Better if we can have a discussion on getting a null value from the `ExternalTreeList`.

Reply:
>  Better if we can have a discussion on getting a null value from the `ExternalTreeList`.
+1